clambcc/clambc-compiler.py

```diff
@@ -836,48 +1080,48 @@ def main():
 #        inFile = os.path.join(os.path.dirname(__file__), 'clambc-compiler-main.c')
 #        args.append(inFile)
 #
-    res = compileFiles(clangLLVM, args, False, bCompiler, options)
+    res = compileFiles(clangLLVM, args, False, False, options)
 
     if not res:
         linkedFile = getLinkedFileName(outFile)
         res = linkFiles(clangLLVM, linkedFile, args, False)
 
     if not res:
         inputSourceFile = getInputSourceFileName(outFile)
-        if bCompiler:
-            f = open(inputSourceFile, "w")
-            f.close()
-        else:
-            res = createInputSourceFile(clangLLVM, inputSourceFile, args, options)
+#        if bCompiler:
+#            f = open(inputSourceFile, "w")
+#            f.close()
+#        else:
+        res = createInputSourceFile(clangLLVM, inputSourceFile, args, options)
 
     if not res:
         optimizedFile = getOptimizedFileName(outFile)
         outFile = getOutfile(options, args)
-        res = optimize(clangLLVM, linkedFile, optimizedFile, outFile, inputSourceFile, bCompiler)
-
-    if not res:
-        if options.genexe:
-
-            #Add the 'main' and all the stuff that clam provides (TODO: make this configurable by the user)
-            mainFile = os.path.join(os.path.dirname(__file__), 'clambc-compiler-main.c')
-            res = compileFile(clangLLVM, mainFile, False, False, options)
-            if res:
-                print("Build FAILED")
-                import pdb ; pdb.set_trace()
-
-            if not res:
-                mainIRFile = getIrFile(mainFile, False)
+        res = optimize(clangLLVM, linkedFile, optimizedFile, outFile, inputSourceFile, False)
 
-                fixFileSize(optimizedFile)
-                fixFileSize(mainIRFile)
-
-                res = linkIRFiles(clangLLVM, optimizedFile, [optimizedFile, mainIRFile])
-
-            bCompiler = True
+#    if not res:
+#        if options.genexe:
+#
+#            #Add the 'main' and all the stuff that clam provides (TODO: make this configurable by the user)
+#            mainFile = os.path.join(os.path.dirname(__file__), 'clambc-compiler-main.c')
+#            res = compileFile(clangLLVM, mainFile, False, False, options)
+#            if res:
+#                print("Build FAILED")
+#                import pdb ; pdb.set_trace()
+#
+#            if not res:
+#                mainIRFile = getIrFile(mainFile, False)
+#
+#                fixFileSize(optimizedFile)
+#                fixFileSize(mainIRFile)
+#
+#                res = linkIRFiles(clangLLVM, optimizedFile, [optimizedFile, mainIRFile])
+#
+#            bCompiler = True
 
-    if not res:
-        if bCompiler:
-            res = genExe(clangLLVM, optimizedFile, outFile)
+#    if not res:
+#        if bCompiler:
+#            res = genExe(clangLLVM, optimizedFile, outFile)
```

Comment:
Do we need to uncomment this? Seems important.

Reply:
Those options were removed.  Was an attempt to build an actual exe that could be more easily debugged without having to run with clam.